flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/KafkaConsumerThread.java

```diff
@@ -80,6 +82,9 @@
 	/** The queue of unassigned partitions that we need to assign to the Kafka consumer. */
 	private final ClosableBlockingQueue<KafkaTopicPartitionState<TopicPartition>> unassignedPartitionsQueue;
 
+	/** The list of partitions to be removed from kafka consumer. */
+	private final List<TopicPartition> partitionsToBeRemoved;
```

Comment:
Should this be Set to facilitate fast lookup ?

Reply:
You are right, a Set should be better for all the calls to the `contains()` method.